src/lib/eina/eina_inline_thread_win32.x

```diff
@@ -0,0 +1,168 @@
+/* EINA - EFL data type library
+ * Copyright (C) 2020 Carlos Ré Signor
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library;
+ * if not, see <http://www.gnu.org/licenses/>.
+ */
+#ifndef EINA_INLINE_THREAD_WIN32_H_
+#define EINA_INLINE_THREAD_WIN32_H_
+
+
+
+typedef struct _Eina_Win32_Thread_Func
+{
+  void *data;
+  void *(*func)(void *data);
+
+} Eina_Win32_Thread_Func;
+
+typedef struct _Eina_Win32_Thread_Attr
+{
+  LPSECURITY_ATTRIBUTES lpThreadAttributes;
+  SIZE_T dwStackSize;
+  DWORD dwCreationFlags;
+} Eina_Win32_Thread_Attr;
+
+static inline void *
+_eina_thread_join(Eina_Thread t)
+{
+
+  int ret = (int)WaitForSingleObject(t, INFINITE);
+
+  if (ret != 0) return ret;
+  return NULL;
+}
+
+DWORD WINAPI _eina_thread_func(void *params)
+{
+  return (DWORD)((Eina_Win32_Thread_Func *)params)->func((void *)((Eina_Win32_Thread_Func *)params)->data);
+}
+
+static inline void _eina_thread_set_priority(Eina_Thread_Priority prio, Eina_Thread *t)
+{
+
+  int nPriority = THREAD_PRIORITY_NORMAL;
+
+  switch (prio)
+    {
+      case EINA_THREAD_URGENT:
+        nPriority = THREAD_PRIORITY_HIGHEST;
+        break;
+      case EINA_THREAD_NORMAL:
+        nPriority = THREAD_PRIORITY_NORMAL;
+        break;
+      case EINA_THREAD_BACKGROUND:
+        nPriority = THREAD_PRIORITY_BELOW_NORMAL;
+        break;
+      case EINA_THREAD_IDLE:
+        nPriority = THREAD_PRIORITY_IDLE;
+        break;
+    }
+
+  SetThreadPriority((HANDLE)*t, nPriority);
+}
```

Comment:
```suggestion
static inline void _eina_thread_set_priority(Eina_Thread_Priority prio, Eina_Thread *t)
{

   int nPriority = THREAD_PRIORITY_NORMAL;

   switch (prio)
     {
      case EINA_THREAD_URGENT:
        nPriority = THREAD_PRIORITY_HIGHEST;
        break;
      case EINA_THREAD_NORMAL:
        nPriority = THREAD_PRIORITY_NORMAL;
        break;
      case EINA_THREAD_BACKGROUND:
        nPriority = THREAD_PRIORITY_BELOW_NORMAL;
        break;
      case EINA_THREAD_IDLE:
        nPriority = THREAD_PRIORITY_IDLE;
        break;
     }

   SetThreadPriority((HANDLE)*t, nPriority);
}
```

Reply:
and maybe set normal as a default priority

Reply:
Ok. @JPTIZ has said to initialize nPriority instead of do a default case in switch, to avoid warnings.